src/ffi.rs

```diff
@@ -0,0 +1,485 @@
+//! # BLS12-377 FFI Bindings for Blind Threshold Signatures.
+use rand_chacha::ChaChaRng;
+use rand_core::{RngCore, SeedableRng};
+
+use crate::{
+    curve::zexe::PairingCurve as Bls12_377,
+    group::{Element, Encodable, Point},
+    poly::Poly,
+    sig::{
+        blind::{BG1Scheme, Token},
+        tblind::G1Scheme,
+        Blinder, Scheme, SignatureScheme, ThresholdScheme,
+    },
+    Index, Share,
+};
+
+use bls_crypto::ffi::Buffer;
+
+// TODO(gakonst): Make these bindings more generic. Maybe a macro is needed here since
```

Comment:
@dignifiedquire you might be interested

Reply:
Thanks, I will take a look at this once we cross that bridge for go-drand. cc @nikkolasg